File: classes/cs350/hw/hw03/C#/Member.cs
/* Nik Ramsey
 * Felicia N'tui
 * CMPS 350: Programming Languages
 * 
 * Contents: member class definition and all derived classes.
 */

using System;
using CS350HW3;

namespace CS350HW3{
    public class Member {
	    public int ID;
	    protected string fName, lName;
	    protected static Random r = new Random();
	
       	public Member() { generate(); }

	    public virtual void generate() {
	        ID = r.Next(89999999) + 10000000; // sets ID between 10000000 and 99999999
	        fName = Names.firstName[r.Next() % Names.firstName.Length];
	        lName = Names.lastName[r.Next() % Names.lastName.Length];
	    }

	    // Formats class output into list items
	    // label indicates whether or not to instance is only a member
	    public override string ToString() { return ToString(true); }
	    public virtual string ToString( bool label = true )
	    {
	        return String.Format("{0} | {1:D3}-{2:D2}-{3:D3} | {4,11} {5,-11} | ",
		        label?"MEM":"", ID/100000, ID/1000%100, ID%1000, 
		        fName, lName);
	    }
    }
}

File: classes/cs350/hw/hw03/Java/Member.java
import java.util.Random;

class Member implements Names{
	 //Data Members
	 protected int ID;
	 protected String fName;
	 protected String lName;

	 //Default Constructor
	 public Member(){
		  this.generate();
	 }

	 //Copy Constructor
	 public Member(Member aMember){
		  this.setID(aMember.id());
		  this.setFirst(aMember.first());
		  this.setLast(aMember.last());
	 }

	 protected void generate(){
		  Random generator = new Random();
		  int indexFN = Math.abs(generator.nextInt() % firstName.length);
		  int indexLN = Math.abs(generator.nextInt() % lastName.length);
		  this.setID(Math.abs(generator.nextInt() % 10000));
		  this.setFirst(firstName[indexFN]);
		  this.setLast(lastName[indexLN]);
	 }

	 @Override public String toString(){
		  return this.toString(true);
	 }

	 public String toString(boolean test){
		  String rtn = "";
		  if(test){
				rtn = "MEM ";
		  }
		  rtn += this.id() + " " + this.last() + " " + this.first();
		  return rtn;
	 }

	 public Member setID(int id){
		  this.ID = id;
		  return this;
	 }

	 public Member setFirst(String fn){
		  this.fName = fn;
		  return this;
	 }

	 public Member setLast(String ln){
		  this.lName = ln;
		  return this;
	 }

	 public int id(){
		  return this.ID;
	 }

	 public String first(){
		  return this.fName;
	 }

	 public String last(){
		  return this.lName;
	 }
}
